src/XMakeBuildEngine/Microsoft.Build.csproj

```diff
@@ -356,8 +356,8 @@
     <Compile Include="Definition\ResolvedImport.cs" />
     <Compile Include="Definition\SubToolset.cs" />
     <Compile Include="Definition\Toolset.cs" />
-    <Compile Include="Definition\ToolsetConfigurationReader.cs" />
-    <Compile Include="..\Shared\ToolsetElement.cs">
+    <Compile Include="Definition\ToolsetConfigurationReader.cs" Condition="$(FeatureSystemConfiguration) == 'true'"/>
+    <Compile Include="..\Shared\ToolsetElement.cs" Condition="$(FeatureSystemConfiguration) == 'true'">
```

Comment:
I think if I had to pick I would say #if the contents of the file over a condition in the csproj to avoid confusion in VS. If there's a good reason for this though I'm fine with it.